cpg_workflows/stages/large_cohort.py

```diff
@@ -459,10 +478,95 @@ def queue_jobs(self, cohort: Cohort, inputs: StageInput) -> StageOutput | None:
                 str(inputs.as_path(cohort, Combiner, key='vds')),
                 str(inputs.as_path(cohort, SampleQC)),
                 str(inputs.as_path(cohort, Relatedness, key='relateds_to_drop')),
+                str(inputs.as_path(cohort, Ancestry, key='inferred_pop')),
+                str(inputs.as_path(cohort, MakeSiteOnlyVcf, key='ht')),
                 str(self.expected_outputs(cohort)),
                 init_batch_args=init_batch_args,
                 setup_gcp=True,
             ),
         )
 
         return self.make_outputs(cohort, data=self.expected_outputs(cohort), jobs=[j])
+
+
+@stage(required_stages=[Combiner])
+class GenerateCoverageTable(CohortStage):
+    def expected_outputs(self, cohort: Cohort) -> Path:
+        if coverage_version := config_retrieve(['large_cohort', 'output_versions', 'coverage'], default=None):
+            coverage_version = slugify(coverage_version)
+
+        coverage_version = coverage_version or get_workflow().output_version
+        return cohort.analysis_dataset.prefix() / get_workflow().name / coverage_version / 'coverage.ht'
+
+    def queue_jobs(self, cohort: Cohort, inputs: StageInput) -> StageOutput | None:
+        from cpg_workflows.large_cohort import generate_coverage_table
+
+        j = get_batch().new_job(
+            'GenerateCoverageTable',
+            (self.get_job_attrs() or {}) | {'tool': HAIL_QUERY},
+        )
+        j.image(image_path('cpg_workflows'))
+
+        j.command(
+            query_command(
+                generate_coverage_table,
+                generate_coverage_table.calculate_coverage_ht.__name__,
+                str(inputs.as_path(cohort, Combiner, key='vds')),
+                str(self.expected_outputs(cohort)),
+                str(self.tmp_prefix),
+                setup_gcp=True,
+            ),
+        )
+
+        return self.make_outputs(cohort, data=self.expected_outputs(cohort), jobs=[j])
+
+
+# @stage(required_stages=[Frequencies])
```

Comment:
We need this dependency uncommented I think?

Reply:
Or, given the requirement for both exomes and genomes frequencies that will have come from different underlying datasets, does this actually need to be a separate pipeline?

Reply:
We currently provide the input to PrepareBrowserTable via a configuration parameter, rather than taking the output of Frequencies directly - so we don't have a correctly functioning pipeline at the moment, in terms of the dependency chain.

For now, pending a proper refactor of this pipeline into it's appropriate subsections, I would suggest we:
- Remove the dependency on Frequencies, as we need to figure out how to properly handle multiple datasets with varying input types
- Remove PrepareBrowserTable as an end stage in the large_cohort pipeline in main, and instead create a separate pipeline for it.